lisa/nic.py

```diff
@@ -381,14 +417,15 @@ def _get_nic_numa_node(self, name: str) -> int:
             numa = 0
         return numa
 
-    def _get_node_nic_info(self) -> None:
+    def _load_nics(self) -> None:
         # Identify which nics are slaved to master devices.
         # This should be really simple with /usr/bin/ip but experience shows
         # the tool isn't super consistent across distros in this regard
 
-        # use sysfs to gather upper/lower nic pairings and pci slot info
+        # use sysfs to gather synthetic/pci nic pairings and pci slot info
         nic_info_fetch_cmd = "ls -la /sys/class/net/*/lower*/device"
         self._node.log.debug(f"Gathering NIC information on {self._node.name}.")
+        lspci = self._node.tools[Lspci]
```

Comment:
How about use `lspci.get_devices_by_type` to get all nic pcis firstly? The current logic may not be right to load mlx and other unknown nics.
And then query from net devices to map them. It doesn't need to use separated logic to handle mana, when loading nics.

Reply:
This method looks complicated than I imagined. Anyway, it can be refined next time.